clang/lib/CodeGen/CGCall.cpp

```diff
@@ -742,8 +742,9 @@ CodeGenTypes::arrangeLLVMFunctionInfo(CanQualType resultType,
                                       FunctionType::ExtInfo info,
                      ArrayRef<FunctionProtoType::ExtParameterInfo> paramInfos,
                                       RequiredArgs required) {
-  assert(llvm::all_of(argTypes,
-                      [](CanQualType T) { return T.isCanonicalAsParam(); }));
+  if (!getContext().getLangOpts().OpenMP)
```

Comment:
Without this map does not work

Reply:
suspicious, wonder why we differ from nvptx here. Might be address space qualification related